apollo-federation/src/sources/connect/variable.rs

```diff
@@ -275,15 +205,6 @@ pub(crate) struct VariablePathPart<'a> {
     pub(crate) location: Range<usize>,
 }
 
-impl VariablePathPart<'_> {
-    fn into_owned(self) -> VariablePathPart<'static> {
```

Comment:
Since you're removing `into_owned`, this eliminates the need for the use of `Cow` in this file. Just replace these with `&'a str`, and update the places that pass in a `Cow`.